docs/src/code/style-guide.adoc

```diff
@@ -252,16 +282,126 @@ of .c and .h are reserved for plain C. Headers are for class, method,
+though. Still, we need this to shine if we want LinuxCNC to shine and
+for a transfer of our knowledge for the next generation - this seems
+fair to say for a project that was started in the last millenium and few
+individuals buying their own mills/lathes before their hair turns gray.
+
+=== Overall structure of documentation
+
+There are two basic documents, i.e. the
+ * Users' Guide and the
+ * Developers' Guide
+
+All documents created belong to either of these "parental documents"
+and are included from the respective document, either directly or that
+included file includes it.
+
+=== Common challenges
+
+ * header tags block like toc etc
+   
+   Every file should start with a header. This is typically
+   ----
+   :lang: en
+   ----
+   and longer documents may also chose to set
+   ----
+   :toc:
+   ----
+   This should be followed by an anchor for that section or chapter that is
+   represented in that file, like
+   ----
+   [[cha:filename]]
+   ----
+   If the file has a language-extension like _fr then this extension shall not become part of the anchor.
+   For a better consistency, and maintainability, the anchors need to be the same for all langages.
+   If the page needs extras, e.g. for the syntax-highlighting, then such lines may then follow.
+
+ * anchors for titles and other blocks
+
+   If a chapter/section header shall be granted the option to be
+   referenced from another part of the documentation then it needs an achor.
+   The anchor shall be a combination of an indicator of the kind
+   of block that is referenced (cha,sec,fig,tab,...) together with a
+   short name identifying the object.
+   
+ * index entries for titles and other blocks
+
+   ?
+
+ * lists formatting
+
+   ?
+
+ * usage of lists vs titles
+
+   ?
+
+ * use of bold and italic
+ * structure/page splitting guidance
+   - avoid 2k+ char long lines
+ * table formatting
+
+   Tables can be presented with a series of options.
+
+   - header
+
+     With the header-option set, the fist line will be interpreted accordingly.
+
+   - column width
+
+     in ideal column, the column is constituted only by its data. No
+     vertical separators should be required. The horizontal alignment will
+     also not be required since the writing itself is horizontal enough.
+
+     If individual fields have too much text then a line-break within that
+     field should be provoked.
+
+   - what lines are visible
+
+     As motivated above, there should be no grid lines.
+     The top and bottom lines should separate the table from the surrounding text.
+     Another line could be motivated between the header and the main body. 
+
+   - captions
+
+     Asciidoc makes it difficult to nicely prepare captions. At the same time,
+     captions are the only bit of the whole document that even with the first flick
+     through the document will not escape the reader's attention.
+     ----
+     Instruction needs to be added
+     ----
+
+    - reference
+   
```

Comment:
An entry about code blocks is also needed.

Found this https://blog.mrhaki.com/2019/12/awesome-asciidoctor-auto-number-callouts.html?m=1 very good for callouts in code blocks.